testgen/generators.go

```diff
@@ -656,6 +842,55 @@ See https://github.com/ethereum/hive/tree/master/cmd/hivechain/contracts/callenv
 				return nil
 			},
 		},
+		{
+			Name:  "eip-7702-authorization-list-empty",
+			About: "Calls a transaction with empty authorization list and expects an error(eip7702 tx type)",
+			Run: func(ctx context.Context, t *T) error {
+				caller := t.chain.txinfo.EIP7702.Account
+				msg := map[string]any{
+					"from":              caller,
+					"to":                common.Address{0x01},
+					"authorizationList": []map[string]any{},
+				}
+				var got interface{}
+				err := t.eth.Client().CallContext(ctx, &got, "eth_call", msg, "latest")
+				expectedMessage := core.ErrEmptyAuthList.Error()
+				if !strings.Contains(err.Error(), expectedMessage) {
```

Comment:
We cannot check for a specific message here. For the purpose of this test, it is sufficient to check there was an error. The recorded transcript will contain the error code and message, and the system that runs these tests against clients will verify the error.